nodes/core/io/21-httpin.js

```diff
@@ -134,6 +159,16 @@ module.exports = function(RED) {
                         }
                         msg.res.set('content-length', len);
                     }
+                    
+                    // Deliberately setting AFTER content-length has been calculated as these fields will be deleted by middleware!
+                    if(RED.settings.metricsOn && RED.settings.metricsOn === true) {
+                        // populating a field on res so that the on headers callback can call metrics properly
+                        msg.res._metrics = {};
+                        msg.res._metrics._messageUuid = msg._messageUuid;
+                        msg.res._metrics._type = node.type;
+                        msg.res._metrics._responsenodeid = node.id;
```

Comment:
I log both the "input node" and "response node" UUIDs in the metrics. I think it's necessary.